api/include/opentelemetry/config.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#ifndef __has_include
+#  define __has_include(x) 0
+#endif
+
+#if !defined(__GLIBCXX__) || __has_include(<codecvt>)  // >= libstdc++-5
```

Comment:
Comment applies to WIndows / msvc: ___has_include_ is (technically) available only since C++17 (vs2017+), but **codecvt** was available since C++11 (vs2015). Maybe we could include codecvt unconditionally, assuming that the compiler, as a minimum, is C++11 and above -compatible?

Reply:
I believe codecvt is missing on the older versions of gcc, but I'm sure there's a way to tweak the condition to have it work.

MSVC won't have __GLIBCXX__ defined, though, right?

Could we add the oldest mscv version we want to support to the CI? Currently it's testing against vs2019
https://github.com/open-telemetry/opentelemetry-cpp/blob/master/.circleci/config.yml#L120-L128

Reply:
I think we might want to vs2017. Something like:

```
#if __cplusplus >= 201103L
...
#endif
```
should work.

My point is - maybe we don't even need to check if we just say that the compiler must be C++11-compliant? That automatically excludes the older gcc from the list of supported compilers.

Reply:
Because people are still using older versions of gcc and it's easy enough to make workarounds for the parts of c++11 that they don't support.